build.zig

```diff
@@ -72,10 +88,49 @@ pub fn build(b: *std.Build) void {
 
     // @ivanv: fix all of our libvmm includes! This is a mess!
     libvmm.addIncludePath(b.path("src"));
-    libvmm.addIncludePath(b.path("src/util/"));
+    libvmm.addIncludePath(b.path("src/util"));
+    libvmm.addIncludePath(b.path("src/virtio"));
     libvmm.addIncludePath(b.path("src/arch/aarch64"));
-    libvmm.addIncludePath(b.path("src/arch/aarch64/vgic/"));
+    libvmm.addIncludePath(b.path("src/arch/aarch64/vgic"));
+    libvmm.addIncludePath(sddf_dep.path("include"));
     libvmm.addIncludePath(.{ .cwd_relative = libmicrokit_include });
+    libvmm.linkLibrary(sddf_dep.artifact("util"));
+    libvmm.linkLibrary(sddf_dep.artifact("util_putchar_debug"));
+
+    libvmm.installHeadersDirectory(b.path("src"), "", .{});
+    libvmm.installHeadersDirectory(b.path("src/util"), "", .{});
+    libvmm.installHeadersDirectory(b.path("src/virtio"), "", .{});
+    libvmm.installHeadersDirectory(b.path("src/arch/aarch64"), "", .{});
+    libvmm.installHeadersDirectory(b.path("src/arch/aarch64/vgic"), "vgic", .{});
 
     b.installArtifact(libvmm);
+
+    var target_userlevel_query = target.query;
+    target_userlevel_query.os_tag = .linux;
+    target_userlevel_query.abi = .musl;
+    const target_userlevel = b.resolveTargetQuery(target_userlevel_query);
```

Comment:
This is a bit weird, we should just construct a new target query instead of copying and modifying the current one.

Reply:
i had in mind that we don't want the outerlevel zig build to know anything about platform specific stuff. otherwise we'll need to pass in the board we're using, or the specific architecture we're working with to construct that target. in the end we only care about changing the os_tag and abi anyway and not the architecture. happy to do either